browser/src/Services/Explorer/ExplorerSplit.tsx

```diff
@@ -74,12 +78,60 @@ export class ExplorerSplit {
         this._store.dispatch({ type: "LEAVE" })
     }
 
+    public moveFileOrFolder = (source: Node, dest: Node): void => {
+        let folderPath
+        let sourcePath
+
+        if (source.type === "folder" && dest.type === "folder") {
+            return this.moveFolder(source, dest)
+        }
+
+        if (dest.type === "file") {
+            const parent = this.findParentDir(dest.id)
+            folderPath = parent
+        } else if (dest.type === "container") {
+            folderPath = dest.name
+        } else {
+            folderPath = dest.folderPath
+        }
+
+        if (folderPath && source && source.type === "file" && source.filePath) {
+            sourcePath = source.filePath
+        } else if (source && source.type === "folder" && folderPath) {
+            sourcePath = source.folderPath
+        }
+
+        Log.info(`moving: ${sourcePath} to ${folderPath}`)
+        mv(sourcePath, folderPath)
+        this._store.dispatch({ type: "REFRESH" })
+    }
+
+    public moveFolder = (
+        source: ExplorerSelectors.IFolderNode,
+        destination: ExplorerSelectors.IFolderNode,
+    ) => {
+        if (source.folderPath === destination.folderPath) {
+            return
+        }
+        Log.info(`moving folders: ${source.folderPath} to ${destination.folderPath}`)
+        mv(source.folderPath, destination.folderPath)
```

Comment:
Is `mv` sync? Just wondering if there can be a race for the move call and the refresh action we dispatch.

Reply:
From the docs all the `shelljs` commands I believe are sync they have a mega thread on their repo about adding async support but doesn't exist yet